### PR TITLE
go/keymanager/api: Add signed encrypted master secret

### PR DESCRIPTION
**Pull request by peternose**
Master secret rotation, 2nd part:
- Add signed encrypted master secret.

### CODE REVIEW

File: go/keymanager/api/secret.go [hunk 1]
```diff
@@ -42,38 +106,47 @@ func (s *EncryptedEphemeralSecret) SanityCheck(epoch beacon.EpochTime, reks map[
...
+	// Secret is the encrypted ephemeral secret.
 	Secret EncryptedEphemeralSecret `json:"secret"`
 
 	// Signature is a signature of the ephemeral secret.
 	Signature signature.RawSignature `json:"signature"`
 }
 
 // Verify sanity checks the encrypted ephemeral secret and verifies its signature.
-func (s *SignedEncryptedEphemeralSecret) Verify(epoch beacon.EpochTime, reks map[x25519.PublicKey]struct{}, rak signature.PublicKey) error {
+func (s *SignedEncryptedEphemeralSecret) Verify(epoch beacon.EpochTime, reks map[x25519.PublicKey]struct{}, rak *signature.PublicKey) error {
```
Comment (kostko) [inline go/keymanager/api/secret.go:149]: Why this change?

Comment (peternose) [reply go/keymanager/api/secret.go:149]: I plan to introduce a new (helper) function which fetches `rak` key from node's capabilities. The key will be passed to the `Verifiy` function meaning that the array will be copied at least twice. I thought that was too much and decided to use pointers instead, especially as `rak`s are read-only. Should I revert changes?

**Review by kostko — APPROVED** [review]
